src/Telegram.Bot.Core/Requests/Available Methods/Admin Bot/DeleteChatPhotoRequest.cs

```diff
@@ -8,21 +7,20 @@ namespace Telegram.Bot.Requests
     /// <summary>
     /// Delete a chat photo. Photos can't be changed for private chats. The bot must be an administrator in the chat for this to work and must have the appropriate admin rights.
     /// </summary>
-    [JsonObject(MemberSerialization.OptIn, NamingStrategyType = typeof(SnakeCaseNamingStrategy))]
-    public class DeleteChatPhotoRequest : RequestBase<bool>
+    public sealed class DeleteChatPhotoRequest : ChatIdRequestBase<bool>
     {
         /// <summary>
-        /// Unique identifier for the target chat or username of the target channel (in the format @channelusername)
+        /// Initializes a new request with <see cref="ChatId"/> set to 0
         /// </summary>
-        [JsonProperty(Required = Required.Always)]
-        public ChatId ChatId { get; }
+        public DeleteChatPhotoRequest() : this(0)
+        {
+        }
```

Comment:
There shouldn't be a default ctor here or in any other request. They aren't meant to be serializable, they're library API and have required parameters in ctors without public setters.

Reply:
Some guys may want to implement the bot's logic using json with reflection, like bot constructor. They need to have as comfortable as possible API.

Reply:
With reflection you can do basically everything. If someone uses reflection for that usecase they can add a bit more code to initialize types without default ctors and properties without setters.